### PR TITLE
Allow replacing installed Minio package on Linux

### PR DESCRIPTION
**Pull request by deepashreeraghu**
Allowing replace of minio package for branch 1.1.1

Fixes: https://github.com/facebookincubator/velox/issues/10043

How was this patch tested?
I ran a build with this fix and now it proceeds further and does not fail at `rpm -i` minio step.

### CODE REVIEW

**Comment by deepashreeraghu** [comment]
I am building Gluten, as part of which I would need to build Velox too, as its backend. I have the S3 flag enabled in this process. 
Even though I build it on a clean image/container ( without any Minio in it) - if the build fails and I retry then the setup-adapters.sh tried to do `rpm -i` which fails giving a non-zero exit, leading to the build failure. So I wanted it to replace the package if it exists so that the build proceeds. 

Since it is expected by Velox for a certain version of Minio - do we need a prompt for this ? Anyways it has to replace with this exact version for it to build, right ?

**Comment by majetideepak** [comment]
Minio is only used by Velox tests. It is not required for the build.

**Comment by deepashreeraghu** [comment]
@majetideepak - Updated the code to get user confirmation before replacing the package, as suggested by you. 
Kindly review.

File: scripts/setup-adapters.sh [hunk 1]
```diff
@@ -35,7 +35,13 @@ function install_aws_deps {
   # We need this specific version of Minio for testing.
   if [[ "$OSTYPE" == linux-gnu* ]]; then
     wget https://dl.min.io/server/minio/release/linux-amd64/archive/minio-20220526054841.0.0.x86_64.rpm
-    rpm -i --replacepkgs minio-20220526054841.0.0.x86_64.rpm
+    if yum list installed minio >/dev/null 2>&1;then
+      if  prompt "Do you want to replace minio package with compatible version minio-20220526054841.0.0.x86_64.rpm?"; then
+        rpm -i --replacepkgs minio-20220526054841.0.0.x86_64.rpm 
+      fi
```
Comment (majetideepak) [inline scripts/setup-adapters.sh:41]: Can we add an else here
`echo "Minio version minio-20220526054841.0.0.x86_64.rpm needed for S3 testing is not installed."`

File: scripts/setup-adapters.sh [hunk 2]
```diff
@@ -35,7 +35,13 @@ function install_aws_deps {
   # We need this specific version of Minio for testing.
   if [[ "$OSTYPE" == linux-gnu* ]]; then
     wget https://dl.min.io/server/minio/release/linux-amd64/archive/minio-20220526054841.0.0.x86_64.rpm
-    rpm -i --replacepkgs minio-20220526054841.0.0.x86_64.rpm
+    if yum list installed minio >/dev/null 2>&1;then
+      if  prompt "Do you want to replace minio package with compatible version minio-20220526054841.0.0.x86_64.rpm?"; then
```
Comment (majetideepak) [inline scripts/setup-adapters.sh:39]: let's say `Do you want to replace the installed Minio with version minio-20220526054841.0.0.x86_64.rpm needed for S3 tests?`

Comment (deepashreeraghu) [reply scripts/setup-adapters.sh]: Updated. Please review.

**Review by majetideepak — APPROVED** [review]
Thanks, @deepashreeraghu

**Comment by deepashreeraghu** [comment]
@majetideepak : Please let me know if there is any action that I have to take on this PR, since it is not merged yet.

**Comment by majetideepak** [comment]
@deepashreeraghu No action is required on your side. We are updating the CI images to Centos9 after which the Build & Push Docker images which is currently failing will succeed.

**Comment by deepashreeraghu** [comment]
Thank you @majetideepak

**Comment by majetideepak** [comment]
@deepashreeraghu can you rebase with the main branch? Thanks!

**Comment by deepashreeraghu** [comment]
Rebase completed.

**Review by majetideepak — APPROVED** [review]
Thanks, @deepashreeraghu

**Pull request closed without merging**